client/components/app.jsx

```diff
@@ -0,0 +1,136 @@
+import React, { Component } from 'react';
+import Header from './layout/Header';
+
+export default class TipCalculator extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      tipPercentage: {
```

Comment:
since the decimal and percent values are mathematically related, this could just be an array instead of an object, which might simplify things in some places:

```
this.state = {
tipPercentage: [0.15, 0.16, 0.17, 0.18, 0.19, 0.20]
tipSelected: null,
      checkTotal: '',
      checkTotalSet: false,
      tipTotal: null,
      grandTotal: null,
      fateDecides: false,
    };
```